server/src/main/java/org/elasticsearch/action/bulk/BulkItemResponse.java

```diff
@@ -341,66 +341,41 @@ public String toString() {
         }
     }
 
-    private int id;
+    private final int id;
 
-    private OpType opType;
+    private final OpType opType;
 
-    private DocWriteResponse response;
+    private final DocWriteResponse response;
 
-    private Failure failure;
-
-    BulkItemResponse() {}
+    private final Failure failure;
```

Comment:
This all started with me wanting to make these final to make them easier to reason about.